torch/_ops.py

```diff
@@ -25,6 +25,98 @@ def dl_open_guard():
     if _SET_GLOBAL_FLAGS:
         sys.setdlopenflags(old_flags)
 
+class OpOverload:
+    def __init__(self, overloadpacket, op, schema):
+        self.op = op
+        self.schema = schema
+        self.overloadpacket = overloadpacket
+
+    # it's a no-op since OpOverload object is immutable and must be for a given op overload.
+    def __deepcopy__(self, memo=None):
+        return self
+
+    def __str__(self):
+        return "OpOverload(op='{}.{}', overload='{}')".format(*self.qualified_name().split("::"), self.overload_name())
+
+    def __call__(self, *args, **kwargs):
+        return self.op(*args, **kwargs or {})
+
+    def __getattr__(self, key):
+        return getattr(self.op, key)
+
+    # `my_namespace::my_op`
+    def qualified_name(self):
+        return self.schema.name
+
+    def overload_name(self):
+        return self.schema.overload_name
+
+    # vector of Arguments -- each Argument's name, and type can currently be accessed
+    # it might be useful to make the mutability of the arg accessible (using the alias_info stored in the C++ struct)
+    def returns(self):
+        return self.schema.returns
+
+    def arguments(self):
+        return self.schema.arguments
+
+class OpOverloadPacket():
+    def __init__(self, qualified_op_name, op_name, op):
+        self.qualified_op_name = qualified_op_name
+        self.op_name = op_name
+        self.op = op
+
+    # it's a no-op since OpOverloadPacket object is immutable and must be for a given op.
+    def __deepcopy__(self, memo=None):
+        return self
+
+    def __str__(self):
+        return "OpOverloadPacket(op='{}.{}')".format(*self.qualified_op_name.split("::"))
+
+    def __getattr__(self, key):
+        # It is not a valid op_name when __file__ is passed in
+        if key == '__file__':
+            return 'torch.ops'
+
+        throw_error = False
+
+        try:
+            use_key = "" if key == 'default' else key
+            op_ = torch._C._get_operation_overload(self.qualified_op_name, use_key)
```

Comment:
TODO: disallow access to overloads registered by JIT. cc. @eellison is there a way to find out which overloads/ops are registered by JIT